### PR TITLE
add options for cross compiling

### PR DESCRIPTION
**Pull request by bollenn**
add options for cross compiling

### CODE REVIEW

**Comment by bramburn** [comment]
I've done something similar but using Cmake and vcpkg

File: gpio/Makefile [hunk 1]
```diff
@@ -32,7 +32,19 @@ endif
...
 #DEBUG	= -g -O0
 DEBUG	= -O2
-CC	= gcc
+
+GCC_ARCH := $(shell gcc -dumpmachine)
+
+ifeq ($(OS),Windows_NT)
+	CROSS_COMPILE ?= arm-linux-gnueabihf-
+else
+	ifneq ($(GCC_ARCH),arm-linux-gnueabihf)
```
Comment (CoRfr) [inline gpio/Makefile]: The indentation is a bit odd here, maybe use spaces with something like:

ifeq ($(OS),Windows_NT)
  CROSS_COMPILE ?= arm-linux-gnueabihf-
else
  ifneq ($(GCC_ARCH),arm-linux-gnueabihf)
    CROSS_COMPILE ?= arm-linux-gnueabihf-
  endif
end

Comment (bollenn) [reply gpio/Makefile]: converted tabs to spaces

**Comment by Gadgetoid** [comment]
As mentioned in #73 I think this is now defunct.

Though I will add- I *prefer* CMake

**Pull request closed without merging**